### PR TITLE
Implement Cyclic Learning Rate and Step-wise Learning Rate Scheduler

### PR DESCRIPTION
**Pull request by Franklalalala**
As mentioned in [issue 211], this PR aims to Implement Cyclic Learning Rate and Step-wise Learning Rate Scheduler.

Major changes include:
1. add a Cyclic Learning Rate in tool.py and corresponding args.
2. add step-wise lr update scheme in trainer.py and base_trainer.py. 
3. unit test, here I implemented 4 cases, which are mesh test on two lr scheme, exp and clr, and w/wo iteration-wise lr update

Minor change:
1. docs have been updated in argcheck.py
2. use the 'display_freq' as the tensorboard log interval (per-iteration), instead of using a fixed 25 iteration interval.

### CODE REVIEW

**Comment by Franklalalala** [comment]
the test example has been updated:
the dataset size was reduced from 5MB to 68KB

File: dptb/nnops/trainer.py [hunk 1]
```diff
@@ -129,6 +130,11 @@ def iteration(self, batch, ref_batch=None):
         loss.backward()
         #TODO: add clip large gradient
         self.optimizer.step()
+        if self.update_lr_per_step_flag:
```
[the remark this answers is not in the document]
Comment (Franklalalala) [reply dptb/nnops/trainer.py:133]: 学习率更新需要显式的使用 self.lr_scheduler.step()。添加这个开关可以在每个 iteration 里调用，否则是每个 epoch 调用一次

File: examples/clr_and_per_iter_update/data_10/data.1312.lmdb/data.mdb
[the remark this answers is not in the document]
Comment (Franklalalala) [reply examples/clr_and_per_iter_update/data_10/data.1312.lmdb/data.mdb:1]: done

**Review by floatingCatty — APPROVED** [review]
